Aura

jouni opened the pull request:
I’m working on “all the things” simultaneously in this branch. This PR is mainly just so that others can easily find the work-in-progress version of Aura. Once the parts have settled, I think we should copy the pieces to separate PRs and review them there, instead of in this massive PR.

jouni committed:
incomplete foundations

jouni commented on dev/aura.html:
```diff
@@ -0,0 +1,239 @@
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Aura</title>
+    <link rel="stylesheet" href="/packages/aura/aura.css">
```
Yeah, thanks, I knew that. But I’m using this page to build a simple dummy app so that I can see the components work together, how they come together to form the overall look'n'feel.

jouni committed:
Lucide icons, Inter as an alternative font, layout updates

jouni committed:
wip

jouni committed:
direction-aware back arrow icon

web-padawan committed:
Use relative import for Aura CSS

web-padawan committed:
chore: fix imports order

web-padawan committed:
chore: fix Prettier formatting in Aura dev page

web-padawan commented on dev/aura.html:
```diff
@@ -0,0 +1,265 @@
+<!doctype html>
```
> The details of a contact are kinda small, or? Should this be wider?

It's because `vaadin-master-detail-layout` in the example doesn't have defined detail size / min-size.
So the size is defined by the content, which in this case happens to be the field default width (`12em`).

jouni committed:
Apologies for this commit message :)

jouni committed:
today's updates

jouni committed:
still some more iteration on the color computations

jouni committed:
computed paddings, gaps, font sizes and line heights

jouni committed:
move custom elements into aura subfolder

jouni committed:
remove cascade layer

don't enforce it in the source, leave it up to the developer who imports the stylesheets.

jouni committed:
make app layout radius easily customizable

jouni committed:
blur content behind backdrops

jouni committed:
note about RTL icons

jouni committed:
add more fonts and include licenses

jouni committed:
move Lucide icons and add license

jouni committed:
fix font size calcs

jouni committed:
add a partially working component kitchen sink view

web-padawan committed:
Fix incorrect CSS selector

jouni committed:
add control for surface opacity

jouni committed:
input container module

jouni committed:
fix accent color propagation

jouni committed:
shared overlay styles

jouni committed:
update surface colors across components

jouni committed:
tweak border colors

web-padawan committed:
Fix usage of camel-case properties

jouni committed:
aura avatar

jouni committed:
Update notification.css

jouni committed:
readonly and disabled inputs

jouni committed:
date picker first iteration

jouni committed:
remove unused property declaration

jouni committed:
add todos

jouni committed:
fix: no border-radius on viewport contained MDL backdrop

jouni committed:
improve MDL drawer appearance

jouni committed:
Update surface.css

jouni committed:
better overlay backdrop color in light mode

jouni committed:
message input enhancements

jouni committed:
first button iteration

jouni committed:
avoid styling message-input button

jouni committed:
Update aura.html

jouni committed:
avoid adding box-shadow to tertiary buttons

knoobie commented on packages/aura/src/components/button.css:
```diff
@@ -1,3 +1,33 @@
+Increase padding, but only for buttons that don't have an icon in the default slot.
+Buttons that place an icon in the default slot are assumed to be icon-only buttons.
+*/
+vaadin-button:not(:has(vaadin-icon:not([slot]))) {
```
Just a note: we use no vaadin-icon inside the button, so it might be better to broaden this selector if people e.g. use svg directly or fontawesome's i (font) integration

jouni committed:
fix surface color for message-input and input-container

jouni committed:
fix avatar colors

jouni committed:
card theme

jouni committed:
enhance button hover and active states

jouni committed:
simplify message-input icon-button styles

jouni committed:
Update app-layout.css

jouni committed:
computed radius values

jouni committed:
one of these commits again…

jouni committed:
add generated palette

jouni committed:
badge colors

jouni committed:
aura-overlay-backdrop-filter property

jouni committed:
use background-container for avatar bg

jouni committed:
button fixes

jouni committed:
dialog and confirm-dialog theme

jouni committed:
temporary solution for grid row selection color

jouni committed:
use palette red for input-field-error-color

jouni committed:
limit how fast radius L increase based on on radius-base

jouni committed:
update surface opacity computation and tweak color computation

jouni committed:
algorithm updates to palette, size, surface

rename all aura-*-base properties to aura-base-*

jouni committed:
update demo/dev page

jouni committed:
rename font weight properties

jouni committed:
add "reset all" button to editor

jouni committed:
attempt to improve default contrast for secondary text

jouni committed:
restore primary button text color

jouni committed:
fix HR style for Safari

jouni committed:
aura-surface-opacity-min → aura-overlay-surface-opacity

jouni committed:
initial support for app-layout navbar

jouni committed:
preliminary “user menu” button

jouni committed:
move theme editor button, add a few more example components

jouni committed:
fix text color in Firefox

some bug prevents using the `c` channel inside a `pow()` function in relative colors.

jouni committed:
set text colors on labels and headings

jouni committed:
move aura-view.css out of Aura itself

jouni committed:
Update surface.css

jouni committed:
update TODO comments

jouni committed:
primary button text color

jouni committed:
move app-layout header and footer styling to aura-view.css

jouni committed:
set vaadin-user-color to aura-accent-color

jouni committed:
move demo-specific styles out of aura.css

jouni committed:
move app-layout specific styles out of aura.css

jouni committed:
remove wat button

jouni committed:
Update date-picker.css

jouni committed:
small updates to hr styles on the demo page

jouni committed:
move footer-end and no-frame variants to demo page

jouni committed:
move Lucide mask-image icons to dev page

jouni committed:
fix lucide-icons import path

jouni committed:
add message-list test items

jouni committed:
tweak root MDL sizing

jouni committed:
give more space for badge example

jouni committed:
new chevron down icon (smaller)

jouni committed:
minor palette color tweaks

jouni committed:
menu-bar-button theme

jouni committed:
make input-container surface more prominent

jouni committed:
allow popover to inherit color scheme

jouni committed:
Update side-nav.css

jouni committed:
don't change input button color on hover when readonly or disabled

jouni committed:
fix aura-view header padding logic

jouni committed:
improve color contrast

jouni committed:
tweak padding sizes
